gnome-initial-setup/pages/timezone/gis-timezone-page.c

```diff
@@ -403,6 +410,33 @@ stop_geolocation (GisTimezonePage *page)
   g_clear_object (&priv->geoclue_simple);
 }
 
+static void
+network_changed_cb (GNetworkMonitor *network_monitor,
+                    gboolean network_available,
+                    gpointer data)
+{
+  GisTimezonePage *page = GIS_TIMEZONE_PAGE (data);
+  GisTimezonePagePrivate *priv = gis_timezone_page_get_instance_private (page);
+
+  if (!network_available)
+    return;
+
+  if (priv->current_location != NULL)
+    {
+      g_signal_handler_disconnect (priv->network_monitor,
+                                   priv->network_monitor_handler_id);
+      priv->network_monitor_handler_id = 0;
+      return;
+    }
+
+  /* if we don't have a geoclue client yet, try to get it and the location,
+   * if we already do, then let's just get the location */
+  if (priv->geoclue_client == NULL)
+    get_location_from_geoclue_async (page);
```

Comment:
It strikes me that you could theoretically end up in the following situation:
- network is available; from `gis_timezone_page_constructed()` we call `get_location_from_geoclue_async()`
- network changes not to be available. This callback is called and immediately returns
- network changes to available again. This callback calls again `get_location_from_geoclue_async()` since the first geoclue async call hasn't returned yet and we haven't finished constructing the geoclue client yet

This will lead to the creation of two geoclue client objects, which is not what we want. AFAICS you can avoid that by calling `stop_geolocatiion_if_needed()` in this callback, when `network_available` is not `TRUE`.

Reply:
I think it may actually be safer to stop any ongoing geoclue tasks in the `get_location_from_geoclue_async`.